lambda/aws-contentgen-appsync-lambda/src/image_generator.py

```diff
@@ -0,0 +1,245 @@
+    
+        model_id=input_params['model_config']['modelId']
+      
+        model_kwargs=input_params['model_config']['model_kwargs']
+        params= get_inference_parameters(model_kwargs)
+
+        logger.info(f'SD params :: {params}')
+    
+
+        request = json.dumps({
+                "text_prompts": (
+                        [{"text": input_text, "weight": 1.0}]
+                        + [{"text": negprompt, "weight": -1.0} for negprompt in negative_prompts]
+                ),
+                "cfg_scale":params['cfg_scale'],
+                "seed": params['seed'],
+                "steps": params['steps'],
+                "style_preset": params['style_preset'],
+                "clip_guidance_preset": params['clip_guidance_preset'],
+                "sampler": params['sampler'],
+                "width": params['width'],
+                "height": params['height']
+                })
+                        
+        try:
+            return  self.bedrock_client.invoke_model(
+                modelId= model_id,
+                contentType= "application/json",
+                accept= "application/json",
+                body=request
+                )
+        except Exception as e:
+            logger.error(f"Error occured during generating image:: {e}")
+            return ''
+        
+
+
+    def send_job_status(self,variables):
+ 
+        logger.info(f"send  status variables :: {variables}")
+        query ="""
+        mutation updateGenerateImageStatus  {
+                updateGenerateImageStatus (status: \"$status\", jobid: \"$jobid\", image_path: \"$image_path\", input_text: \"$input_text\", filename: \"$filename\", message: \"$message\") 
+                {
+                    status, 
+                    jobid,
+                    image_path,
+                    filename,
+                    input_text,
+                    message
+                }
+        }
+                """
+
+        query = query.replace("$status", variables['status'])
+        query = query.replace("$jobid", str(variables['jobid']))
+        query = query.replace("$image_path", str(variables['image_path']))
+        query = query.replace("$filename", str(variables['filename']))
+        query = query.replace("$input_text", str(variables['input_text']))
+        query = query.replace("$message", str(variables['message']))
+        request = {'query':query}
+
+        logger.info(request)
+
+        GRAPHQL_URL = os.environ['GRAPHQL_URL']
+        HEADERS={
+            "Content-Type": "application/json",
+        }
+        
+        responseJobstatus = reqs.post(
+            json=request,
+            url=GRAPHQL_URL,
+            headers=HEADERS,
+            auth=aws_auth_appsync,
+            timeout=10
+        )
+        logger.info('res :: {}',responseJobstatus)
+
+
+def get_inference_parameters(model_kwargs):
+      """ Read inference parameters and set default values"""
```

Comment:
you can use .get to specify a default value if an item is not in the input dict
that will avoid the multiple if/else